fvm/transactionPayerBalanceChecker.go

```diff
@@ -12,6 +12,69 @@ import (
 
 type TransactionPayerBalanceChecker struct{}
 
+const VerifyPayerBalanceResultTypeCanExecuteTransactionFieldName = "canExecuteTransaction"
+const VerifyPayerBalanceResultTypeRequiredBalanceFieldName = "requiredBalance"
+const VerifyPayerBalanceResultTypeMaximumTransactionFeesFieldName = "maximumTransactionFees"
+
+var VerifyPayerBalanceResultType = &cadence.StructType{
+	Fields: []cadence.Field{
+		{
+			Identifier: VerifyPayerBalanceResultTypeCanExecuteTransactionFieldName,
+			Type:       cadence.BoolType,
+		},
+		{
+			Identifier: VerifyPayerBalanceResultTypeRequiredBalanceFieldName,
+			Type:       cadence.UFix64Type,
+		},
+		{
+			Identifier: VerifyPayerBalanceResultTypeMaximumTransactionFeesFieldName,
+			Type:       cadence.UFix64Type,
+		},
+	},
+}
+
+// decodeVerifyPayerBalanceResult decodes the VerifyPayerBalanceResult struct
+// https://github.com/onflow/flow-core-contracts/blob/7c70c6a1d33c2879b60c78e363fa68fc6fce13b9/contracts/FlowFees.cdc#L75
+func decodeVerifyPayerBalanceResult(resultValue cadence.Value) (
+	canExecuteTransaction cadence.Bool,
+	requiredBalance cadence.UFix64,
+	maximumTransactionFees cadence.UFix64,
+	err error,
+) {
+	result, ok := resultValue.(cadence.Struct)
+	if !ok {
+		return false, 0, 0, fmt.Errorf("invalid VerifyPayerBalanceResult value: not a struct")
+	}
+
+	fields := cadence.FieldsMappedByName(result)
```

Comment:
IDK – I left original logic as-is, and only adjusted the field extraction. I did not add additional checks because IDK about the business logic.

If cases like this need additional assertions, please open a separate issue and maybe address them on master (this branch only contains changes for Cadence 1.0)